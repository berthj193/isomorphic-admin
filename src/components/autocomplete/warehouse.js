import React from 'react';
import PropTypes from 'prop-types';
import withStoresManager from '../apolloProviders/withStoresManager';
import { getAutocompleteList as getWarehouseAutocompleteList } from '../../graphql/warehouses';
import Select, { SelectOption } from '../uielements/select';

export class WarehouseAutocomplete extends React.Component {
  static propTypes = {
    warehouses: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      name: PropTypes.string,
    })),
  };

  static defaultProps = {
    warehouses: [],
  };

  dataSource = (this.props.warehouses || []).map(({ id, name }) =>
    <SelectOption value={id} key={id}>{`${name}`}</SelectOption>);

  render() {
    return (
      <Select
        showSearch
        {...this.props}
      >
        {this.dataSource}
      </Select>
    );
  }
}

export default withStoresManager(WarehouseAutocomplete, [
  {
    query: getWarehouseAutocompleteList,
  },
]);
