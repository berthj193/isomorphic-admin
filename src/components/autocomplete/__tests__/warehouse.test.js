import React from 'react';
import { WarehouseAutocomplete } from '../warehouse';
import { shallow } from 'enzyme';

describe('warehouse autocomplete component', () => {
  const getComponent = (props = {}) => shallow(<WarehouseAutocomplete {...props} />);
  const dataObj = {
    id: '1',
    name: 'foo',
  };

  it('datasource should return empty array when there are no elements', () => {
    const instance = getComponent().instance();
    expect(instance.dataSource.length).toBe(0);
  });

  it('datasource should return one element', () => {
    const instance = getComponent({
      warehouses: [dataObj],
    }).instance();

    expect(instance.dataSource).toMatchSnapshot();
  });
});
