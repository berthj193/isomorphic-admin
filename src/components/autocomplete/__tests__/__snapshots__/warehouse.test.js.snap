// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`warehouse autocomplete component datasource should return one element 1`] = `
Array [
  <Option
    value="1"
>
    foo
</Option>,
]
`;
